modules/shortcodes/gist.php

```diff
@@ -1,58 +1,91 @@
-	wp_enqueue_script( 'jetpack-gist-embed', plugins_url( 'js/gist.js', __FILE__ ), array( 'jquery' ), false, true );
-
 	// Let the shortcode callback do all the work
-	return github_gist_shortcode( $attr, $url );
+	return github_gist_shortcode( $matches, $url );
 }
 
+/**
+ * Callback for gist shortcode.
+ *
+ * @since 2.8.0
+ *
+ * @param array  $atts Attributes found in the shortcode.
+ * @param string $content Content enclosed by the shortcode.
+ *
+ * @return string The gist HTML.
+ */
 function github_gist_shortcode( $atts, $content = '' ) {
 
-	if ( empty( $atts[0] ) && empty( $content ) )
+	if ( empty( $atts[0] ) && empty( $content ) ) {
 		return '<!-- Missing Gist ID -->';
+	}
 
 	$id = ( ! empty( $content ) ) ? $content : $atts[0];
 
 	// Parse a URL
-	if ( ! is_numeric( $id ) )
+	if ( ! is_numeric( $id ) ) {
 		$id = preg_replace( '#https?://gist.github.com/([a-zA-Z0-9]+)#', '$1', $id );
+	}
 
-	if ( ! $id )
+	if ( ! $id ) {
 		return '<!-- Invalid Gist ID -->';
+	}
 
-	if ( ! empty( $atts['file'] ) ) {
-		$file = '?file=' . urlencode( $atts['file'] );
+	wp_enqueue_script( 'jetpack-gist-embed', plugins_url( 'js/gist.js', __FILE__ ), array( 'jquery' ), false, true );
+
+	if ( false !== strpos( $id, '#file-' ) ) {
+		// URL points to a specific file in the gist
+		$id = str_replace( '#file-', '.json?file=', $id );
+		$id = preg_replace( '/\-(?!.*\-)/', '.', $id );
 	} else {
-		$file = '';
+		$file = ( ! empty( $atts['file'] ) ) ? '?file=' . urlencode( $atts['file'] ) : '';
+		// URL points to the entire gist
+		$id .= ".json$file";
 	}
 
-	$embed_url = "{$id}.json" . $file;
-
 	// inline style to prevent the bottom margin to the embed that themes like TwentyTen, et al., add to tables
-	$return = '<style>.gist table { margin-bottom: 0; }</style>' .
-			  '<div class="gist-oembed" data-gist="' . esc_attr( $embed_url ) . '"></div>';
-
-	if ( isset( $_POST[ 'type' ]) && 'embed' === $_POST[ 'type' ] &&
-			isset( $_POST[ 'action' ] ) && 'parse-embed' === $_POST['action'] ) {
+	$return = '<style>.gist table { margin-bottom: 0; }</style><div class="gist-oembed" data-gist="' . esc_attr( $id ) . '"></div>';
 
-		return github_gist_simple_embed( $id, $file );
+	if ( isset( $_POST[ 'type' ]) && 'embed' === $_POST[ 'type' ] && isset( $_POST[ 'action' ] ) && 'parse-embed' === $_POST['action'] ) {
+		return github_gist_simple_embed( $id );
 	}
 
-	return  $return;
+	return $return;
 }
 
-function github_gist_simple_embed( $id, $file ) {
-	$embed_url = $id . '.js' . $file;
-
-	return '<script type="text/javascript" src="//gist.github.com/' . $embed_url . '"></script>';
+/**
+ * Use script tag to load shortcode in editor.
+ *
+ * @since 3.9.0
+ *
+ * @param string $id The ID of the gist.
+ *
+ * @return string
+ */
+function github_gist_simple_embed( $id ) {
+	$id = str_replace( 'json', 'js', $id );
+	return '<script type="text/javascript" src="//gist.github.com/' . $id . '"></script>';
```

Comment:
@kraftbj good catch, thanks, it's fixed now.

@CFXd I'll be checking the issue with ssl asap and will let you know.